src/main/java/bio/terra/service/iam/sam/SamIam.java

```diff
@@ -419,20 +419,40 @@ public static DataRepoException convertSAMExToDataRepoEx(ApiException samEx) {
 
         switch (samEx.getCode()) {
             case HttpStatusCodes.STATUS_CODE_BAD_REQUEST: {
-                return new IamBadRequestException(samEx);
+                return new IamBadRequestException(
+                    "Invalid auth request",
+                    samEx,
+                    Collections.singletonList(samEx.getResponseBody())
+                );
             }
             case HttpStatusCodes.STATUS_CODE_UNAUTHORIZED: {
-                return new IamUnauthorizedException(samEx);
+                return new IamUnauthorizedException(
+                    "Unauthorized access",
+                    samEx,
+                    Collections.singletonList(samEx.getResponseBody())
```

Comment:
I'm not so sure about this. The effect here will not only be to put the SAM response into our logs, but to return it to the caller. I think that might exposes too much information.

Reply:
Yeah, I wasn't sure about this. In testing locally, it looked like the errorDetails never got returned to the client and only used when printing the logs.  I'm happy to change this to log in the Sam exception converter.  I was just trying to log closer to where the exception was getting used for a little more context.

Reply:
It does get returned to the callers. We don't use it in most cases, but there are specific errors where we use it. For example, IngestValidateRefsStep we return invalid references in it.